scripts/cddl_gen.py

```diff
@@ -1112,6 +1112,8 @@ def single_func_prim(self, access):
             arg = tmp_str_or_null(self.value)
         elif self.type == "BOOL":
             arg = min_bool_or_null(self.value)
+        elif self.type in ["UINT", "INT", "NINT", "FLOAT"] and mode == "decode":
```

Comment:
Is this in the correct commit?

Reply:
Yes, this is the "not use temp var" part, since before, it would fall through to the else below, which uses the temp var. The temp var was necessary to be able to pass literals as pointers. Passing the literals as values instead uses less code.

    uintx32_expect(p_state, (tmp_value = 3, &tmp_value));

turns into 

    uintx32_expect(p_state, 3));